### PR TITLE
Fix generation of API reference

### LINKED ISSUE #2032: Changes to API comments aren't regenerated by generate-apiref
**What would you like to be cleaned**:

Currently, the changes in api comments are not regenerated by `generate-apiref`, discovered in https://github.com/kubernetes-sigs/kueue/pull/2025#pullrequestreview-2014845838

When running locally it fails like this:
```sh
> make generate-apiref
cd  /usr/local/google/home/michalwozniak/go/src/sigs.k8s.io/kueue/site/genref/ && /usr/local/google/home/michalwozniak/go/src/sigs.k8s.io/kueue/bin/genref  -o /usr/local/google/home/michalwozniak/go/src/sigs.k8s.io/kueue/site/content/en/docs/reference
I0422 16:40:45.419533 2863192 main.go:142] Parsing go packages in sigs.k8s.io/kueue/apis/kueue/v1beta1
E0422 16:40:45.426941 2863192 main.go:364] "cannot process API path" err="no loaded build package for sigs.k8s.io/kueue/apis/kueue/v1beta1"
I0422 16:40:45.426975 2863192 main.go:142] Parsing go packages in sigs.k8s.io/kueue/apis/config/v1beta1
E0422 16:40:45.434552 2863192 main.go:364] "cannot process API path" err="no loaded build package for sigs.k8s.io/kueue/apis/config/v1beta1"
I0422 16:40:45.434580 2863192 main.go:142] Parsing go packages in sigs.k8s.io/kueue/apis/kueue/v1alpha1
E0422 16:40:45.440996 2863192 main.go:364] "cannot process API path" err="no loaded build package for sigs.k8s.io/kueue/apis/kueue/v1alpha1"
```

Using git bisect I found this PR as the culprit: https://github.com/kubernetes-sigs/kueue/pull/1884.

**Why is this needed**:

To correctly generate API reference.

### PR DESCRIPTION
**Pull request by alculquicondor**
#### What type of PR is this?

/kind bug
/kind documentation

#### What this PR does / why we need it:

Fix automatic generation of apiref.

#1884 made the site its own go module, limiting the ability for the tool to find the packages defined in the root module.

#### Which issue(s) this PR fixes:

Fixes #2032

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -498,4 +498,4 @@ cluster-autoscaler-crd:
 
 .PHONY: generate-apiref
 generate-apiref: genref
-	cd  $(PROJECT_DIR)/site/genref/ && $(GENREF)  -o $(PROJECT_DIR)/site/content/en/docs/reference
+	cd  $(PROJECT_DIR)/hack/genref/ && $(GENREF)  -o $(PROJECT_DIR)/site/content/en/docs/reference
```
Comment (tenzen-y) [inline Makefile:501]:
```suggestion
	cd  $(PROJECT_DIR)/hack/genref/ && $(GENREF) -o $(PROJECT_DIR)/site/content/en/docs/reference
```

:)

**Review by tenzen-y — COMMENTED** [review]
I left a comment for a nit.

File: Makefile (hunk 1 above)
Comment (alculquicondor) [reply Makefile:501]: hehe very important :)

Comment (alculquicondor) [reply Makefile]: It seems that I forgot to push

**Comment by tenzen-y** [comment]
/lgtm
/approve

/hold
until verified if the `pull-kueue-verify-main` succeeded.

**Comment by tenzen-y** [comment]
🎉 
/hold cancel

> cd /home/prow/go/src/sigs.k8s.io/kueue/hack/genref/ && /home/prow/go/src/sigs.k8s.io/kueue/bin/genref -o /home/prow/go/src/sigs.k8s.io/kueue/site/content/en/docs/reference
I0422 17:39:48.364076   24429 main.go:142] Parsing go packages in sigs.k8s.io/kueue/apis/kueue/v1beta1
I0422 17:39:52.965262   24429 main.go:325] Output written to /home/prow/go/src/sigs.k8s.io/kueue/site/content/en/docs/reference/kueue.v1beta1.md
I0422 17:39:52.965295   24429 main.go:142] Parsing go packages in sigs.k8s.io/kueue/apis/config/v1beta1
I0422 17:40:32.327872   24429 main.go:325] Output written to /home/prow/go/src/sigs.k8s.io/kueue/site/content/en/docs/reference/kueue-config.v1beta1.md
I0422 17:40:32.327954   24429 main.go:142] Parsing go packages in sigs.k8s.io/kueue/apis/kueue/v1alpha1
I0422 17:40:36.218520   24429 main.go:325] Output written to /home/prow/go/src/sigs.k8s.io/kueue/site/content/en/docs/reference/kueue-alpha.v1alpha1.md

https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_kueue/2034/pull-kueue-verify-main/1782462383894040576#1:build-log.txt%3A422

**Pull request merged**